conformance/echo-basic/echo-basic.go

```diff
@@ -200,12 +208,96 @@ func runH2CServer(h2cPort string, errchan chan<- error) {
 	}
 }
 
+// Channel variable to store the SNI retrieved in runBackendTLS handler func.
+var sniChannel = make(chan string)
```

Comment:
```suggestion
var sniChannel = make(chan string, 1)
```
The channel needs to be non-blocking in order for this to work, but I still have some doubts about whether it's truly necessary (see https://github.com/kubernetes-sigs/gateway-api/pull/3212/files#r2127988955).

It might also be possible to use the `context.Context` instead of a channel, though I’m not exactly sure how that would look in practice, especially since the handshake and the request generally operate with different contexts.